test/integration/virtualnode/virtualnode_test.go

```diff
@@ -0,0 +1,150 @@
+package virtualnode_test
+
+import (
+	"context"
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	appmesh "github.com/aws/aws-app-mesh-controller-for-k8s/apis/appmesh/v1beta2"
+	"github.com/aws/aws-app-mesh-controller-for-k8s/pkg/algorithm"
+	"github.com/aws/aws-app-mesh-controller-for-k8s/test/framework"
+	"github.com/aws/aws-app-mesh-controller-for-k8s/test/framework/manifest"
+	"github.com/aws/aws-app-mesh-controller-for-k8s/test/framework/utils"
+	"github.com/aws/aws-app-mesh-controller-for-k8s/test/integration/mesh"
+	"github.com/aws/aws-app-mesh-controller-for-k8s/test/integration/virtualnode"
+
+	//"github.com/aws/aws-sdk-go/aws"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ = Describe("VirtualNode", func() {
+
+	var (
+		ctx context.Context
+		f   *framework.Framework
+	)
+
+	BeforeEach(func() {
+		ctx = context.Background()
+		if f == nil {
+			f = framework.New(framework.GlobalOptions)
+		}
+
+		if f.Options.ControllerImage != "" {
+			By("Reset cluster with default controller", func() {
+				f.HelmManager.ResetAppMeshController()
+			})
+		}
+		if f.Options.InjectorImage != "" {
+			By("Reset cluster with default injector", func() {
+				f.HelmManager.ResetAppMeshInjector()
+			})
+		}
+	})
+
+	Context("Virtual Node create scenaries", func() {
```

Comment:
Plan to add Update/Delete scenarios via a new PR? Good to validate these https://github.com/aws/aws-app-mesh-controller-for-k8s/blob/master/webhooks/appmesh/virtualnode_validator.go#L49

Reply:
ThYes, I plan to add these scenarios in a follow up. This is just the baseline to get started on the integration tests